packages/lit-dev-content/src/components/litdev-aside.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {css, html, LitElement} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {greenCheckIcon} from '../icons/green-check-icon.js';
+import {redXIcon} from '../icons/red-x-icon.js';
+import {yellowBangIcon} from '../icons/yellow-bang-icon';
+import {blueInfoIcon} from '../icons/blue-info-icon';
+
+export type AsideVariant = 'positive' | 'negative' | 'warn' | 'info';
+
+@customElement('litdev-aside')
+export class LitDevAside extends LitElement {
+  static styles = css`
+    :host {
+      color: currentColor;
+      display: block;
+      margin: 1em 0;
+    }
+
+    aside {
+      display: flex;
+      border-style: solid;
+      border-width: 1px;
+      border-color: var(--lit-dev-aside-border-color, #ccc);
+      padding: 1em 1em 1em 0em;
+    }
+
+    slot {
+      display: block;
+      flex-grow: 1;
+    }
+
+    svg {
+      width: 1.5em;
+      margin-inline: 1em;
+    }
+
+    :host(:not([no-header])) ::slotted(:first-child) {
+      font-weight: bold;
+    }
+
+    ::slotted(*) {
+      margin: 0;
+    }
+  `;
+
+  @property({type: String})
+  type: AsideVariant = 'info';
+
+  @property({type: Boolean, reflect: true, attribute: 'no-header'})
+  noHeader = false;
+
+  render() {
+    return html`
+      <aside>
+        <div>${this._renderIcon()}</div>
+        <slot></slot>
+      </aside>
+    `;
+  }
+
+  private _renderIcon() {
+    switch (this.type) {
+      case 'positive':
+        return greenCheckIcon;
+      case 'negative':
+        return redXIcon;
+      case 'warn':
+        return yellowBangIcon;
+      case 'info':
+        return blueInfoIcon;
+    }
```

Comment:
I'm torn here because I'm a fan of HTML "failing successfully", but also you don't write these components in a place where `lit-plugin` is available and you have an increased chance to write a wrong `type`. Though, we don't really have a CI check to check for thrown exceptions, so it might just pass this error onto the user

Reply:
It seems valid to not want to throw when the type is coming from a user supplied attribute, not something hard coded. Perhaps something like this? (not sure about log level, warn vs error)
```ts
default:
  const exhaustiveCheck: never = this.type;
  logger.warn(`Received unexpected type for <litdev-aside>: ${exhaustiveCheck}`);
```
Then TS will make sure we've handled all expected cases when authoring, without runtime throwing.